guestbook/README.md

```diff
@@ -4,704 +4,306 @@
-
-The traffic flow from slaves to masters can be described in two steps:
-
-  - A *redis slave* will connect to `port` on the *redis master Service*
-  - Traffic will be forwarded from the Service `port` (on the Service node) to the `targetPort` on the pod that the Service listens to.
-
-For more details, please see [Connecting applications](http://kubernetes.io/docs/user-guide/connecting-applications/).
-
-#### Finding a Service
-
-Kubernetes supports two primary modes of finding a Service — environment variables and DNS.
+        kubectl apply -f redis-master-service.yaml
 
+{% include code.html language="yaml" file="redis-master-service.yaml" ghlink="/docs/tutorials/docs/tutorials/stateless-application/redis-master-service.yaml" %}
 
-##### Environment variables
+**Note:** This manifest file creates a Service named `redis-master` with a set of labels that match the labels previously defined, so the Service routes network traffic to the Redis master Pod.   
+{: .note}
 
-The services in a Kubernetes cluster are discoverable inside other containers via [environment variables](https://kubernetes.io/docs/concepts/services-networking/service/#environment-variables).
+{:start="2"}
+2. Query the list of Services to verify that the Redis Master Service is running:
 
-##### DNS service
+        kubectl get service
 
-An alternative is to use the [cluster's DNS service](https://kubernetes.io/docs/concepts/services-networking/service/#dns), if it has been enabled for the cluster.  This lets all pods do name resolution of services automatically, based on the Service name.
+   The response should be similar to this:
 
-This example has been configured to use the DNS service by default.
+        NAME           CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
+        kubernetes     10.0.0.1     <none>        443/TCP    1m
+        redis-master   10.0.0.151   <none>        6379/TCP   8s
 
-If your cluster does not have the DNS service enabled, then you can use environment variables by setting the
-`GET_HOSTS_FROM` env value in both
-[redis-slave-deployment.yaml](https://git.k8s.io/examples/guestbook/redis-slave-deployment.yaml) and [frontend-deployment.yaml](https://git.k8s.io/examples/guestbook/frontend-deployment.yaml)
-from `dns` to `env` before you start up the app.
-(However, this is unlikely to be necessary. You can check for the DNS service in the list of the cluster's services by
-running `kubectl --namespace=kube-system get rc -l k8s-app=kube-dns`.)
-Note that switching to env causes creation-order dependencies, since Services need to be created before their clients that require env vars.
+## Start up the Redis Slaves
 
-#### Create a Deployment
+Although the Redis master is a single pod, you can make it highly available to meet traffic demands by adding replica Redis workers.
```

Comment:
slave vs worker: please unify the terminology. either rename files (and name: fields) to worker, or just call them slave throughout the tutorial)